armi/reactor/components/component.py

```diff
@@ -232,11 +232,26 @@ def __init__(
         self.temperatureInC = Thot
         self.material = None
         self.setProperties(material)
+        self.tInputWarning(Tinput)
         self.applyMaterialMassFracsToNumberDensities()  # not necessary when duplicating...
         self.setType(name)
         self.p.mergeWith = mergeWith
         self.p.customIsotopicsName = isotopics
 
+    def tInputWarning(self, Tinput):
+        """
+        Check whether thermal expansion factor is 0.0% exactly at T=Tinput
+        """
+        expansionFactor = (
+            self.material.linearExpansionPercent(Tc=self.inputTemperatureInC) / 100.0
+        )
+        if not (abs(expansionFactor) < 1.0e-6):
+            runLog.warning(
```

Comment:
Let's make this a single warning so it doesnt show up a million times in the stdout

Reply:
Done